distribution/integration/tests-integration/src/test/java/org/wso2/carbon/integration/tests/carbontools/CarbonServerBasicOperationTestCase.java

```diff
@@ -0,0 +1,226 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.integration.tests.carbontools;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.ContextXpathConstants;
+import org.wso2.carbon.integration.common.exception.CarbonToolsIntegrationTestException;
+import org.wso2.carbon.integration.common.utils.CarbonCommandToolsUtil;
+import org.wso2.carbon.integration.common.utils.CarbonIntegrationBaseTest;
+import org.wso2.carbon.utils.ServerConstants;
+import sun.management.VMManagement;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This class has the test methods for start, stop, restart, dump and run build.xml test cases
+ */
+public class CarbonServerBasicOperationTestCase extends CarbonIntegrationBaseTest {
+
+    private static final Log log = LogFactory.getLog(CarbonServerBasicOperationTestCase.class);
+    private HashMap<String, String> serverPropertyMap = new HashMap<String, String>();
+    private String carbonHome;
+    private AutomationContext context;
+    private final int portOffset = 1;
+    private String processId;
+    Process processStop;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws CarbonToolsIntegrationTestException, XPathExpressionException {
+        serverPropertyMap.put("-DportOffset", Integer.toString(portOffset));
+        context = new AutomationContext("CARBON", "carbon002",
+                                        ContextXpathConstants.SUPER_TENANT,
+                                        ContextXpathConstants.ADMIN);
+        carbonHome = CarbonCommandToolsUtil.getCarbonHome(context);
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server start test")
+    public void testStartCommand() throws CarbonToolsIntegrationTestException,
+                                          NoSuchFieldException, IllegalAccessException {
+        String[] cmdArrayToStart;
+        Process process;
+
+        if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
+            throw new SkipException("Feature --start is not available for windows");
```

Comment:
Explain why we are trowing testNG skip exception here. You need to comment properly.